src/diagonal.works/b6/api/functions/change.go

```diff
@@ -107,6 +108,65 @@ func addRelation(c *api.Context, id b6.RelationID, tags api.TagCollection, membe
 	}, nil
 }
 
+func addCollection(c *api.Context, id b6.CollectionID, tags api.TagCollection, collection api.Collection) (ingest.Change, error) {
+	feature := &ingest.CollectionFeature{
+		CollectionID: id,
+	}
+
+	i := tags.Begin()
+	for {
+		ok, err := i.Next()
+		if err != nil {
+			return nil, err
+		} else if !ok {
+			break
+		}
+		tag, ok := i.Value().(b6.Tag)
+		if !ok {
+			return nil, fmt.Errorf("Expected b6.Tag, found %T", i.Value())
+		}
+		feature.Tags = append(feature.Tags, tag)
+	}
+
+	i = collection.Begin()
+	for {
+		ok, err := i.Next()
+		if err != nil {
+			return nil, err
+		} else if !ok {
+			break
+		}
+
+		feature.CollectionProto.Keys = append(feature.CollectionProto.Keys, toLiteralNodeProto(i.Key()))
+		feature.CollectionProto.Values = append(feature.CollectionProto.Values, toLiteralNodeProto(i.Value()))
+	}
+
+	return &ingest.AddFeatures{
+		Collections: []*ingest.CollectionFeature{feature},
+	}, nil
+}
+
+func toLiteralNodeProto(v interface{}) *pb.LiteralNodeProto {
+	switch x := v.(type) {
+	// TODO: cover all types
```

Comment:
afaic u only did string / identifiable in relations so this is fine to follow up w?

Reply:
n/a